sklearn/_build_utils/openmp_helpers.py

```diff
@@ -59,85 +46,69 @@ def get_openmp_flag(compiler):
-                    flag for flag in extra_preargs
-                    if flag.startswith(('-L', '-Wl,-rpath', '-l'))]
+    extra_postargs = get_openmp_flag
 
-            objects = glob.glob(
-                os.path.join('objects', '*' + ccompiler.obj_extension))
-            ccompiler.link_executable(objects, 'test_openmp',
+    try:
+        output = compile_test_program(code,
                                       extra_preargs=extra_preargs,
-                                      extra_postargs=openmp_flags)
-
-            # Run test program
-            output = subprocess.check_output('./test_openmp')
-            output = output.decode(sys.stdout.encoding or 'utf-8').splitlines()
-
-            # Check test program output
-            if 'nthreads=' in output[0]:
-                nthreads = int(output[0].strip().split('=')[1])
-                openmp_supported = (len(output) == nthreads)
-            else:
-                openmp_supported = False
+                                      extra_postargs=extra_postargs)
 
-        except (CompileError, LinkError, subprocess.CalledProcessError):
+        if 'nthreads=' in output[0]:
+            nthreads = int(output[0].strip().split('=')[1])
+            openmp_supported = len(output) == nthreads
+        else:
             openmp_supported = False
 
-        finally:
-            os.chdir(start_dir)
+    except (CompileError, LinkError, subprocess.CalledProcessError):
+        openmp_supported = False
 
-    err_message = textwrap.dedent(
+    message = textwrap.dedent(
         """
-                            ***
+
+                        ***********
+                        * WARNING *
+                        ***********
 
         It seems that scikit-learn cannot be built with OpenMP support.
 
         - Make sure you have followed the installation instructions:
 
             https://scikit-learn.org/dev/developers/advanced_installation.html
 
-        - If your compiler supports OpenMP but the build still fails, please
-          submit a bug report at:
+        - If your compiler supports OpenMP but you still see this message,
+          please submit a bug report at:
 
             https://github.com/scikit-learn/scikit-learn/issues
 
-        - If you want to build scikit-learn without OpenMP support, you can set
-          the environment variable SKLEARN_NO_OPENMP and rerun the build
-          command. Note however that some estimators will run in sequential
-          mode and their `n_jobs` parameter will have no effect anymore.
+        - The build will continue without OpenMP support. Note however that
+          some estimators will run in sequential mode instead of leveraging
+          thread-based parallelism.
 
                             ***
         """)
 
     if not openmp_supported:
-        raise CompileError(err_message)
+        if os.getenv("SKLEARN_FAIL_NO_OPENMP"):
+            raise CompileError("Failed to build with OpenMP")
```

Comment:
I do point to this address in the warning message. Do you mean to also point there in the error message ? I don't see why because if you explicitly set SKLEARN_FAIL_NO_OPENMP it means you come from this page.

Reply:
I agree, this is a fair point.